### PR TITLE
Improvements and updates

### PR DESCRIPTION
**Pull request by cablespaghetti**
A few improvements and updates to use the latest Node JS and add the interests field we need.

### CODE REVIEW

**Comment by LeanderK** [comment]
Open source magic! Can you take a look at it @jundl77 ?

**Comment by jundl77** [comment]
This looks pretty good to me, I didn't test anything, but if it runs than great. 

**There is a more important issue with this, however.** At the time I built this, there was no DSGVO and what this does is that it pretty much squashes the approval email that MailChimp usually sends to directly add the person to the mailing list. 

This is no longer legal, however. Under the DSGVO, people need to get an approval email asking them to sign up on the mailing list, and they also need to be able to unregister (something which I am not sure how it is handled).

**Review by jundl77 — COMMENTED** [review]
Thanks for taking the time to upgrade our little mail proxy! I hope you find it useful :)

Anyway, this looks pretty good to me, I didn't test anything, but if it runs then great.

**There is a more important issue with this, however.** At the time I built this, there was no DSGVO and  this proxy pretty much squashes the approval email that MailChimp usually sends to directly add the person to the mailing list.

This is no longer legal, however. Under the DSGVO, people need to get an approval email asking them to sign up on the mailing list, and they also need to be able to unregister (something which I am not sure how it is handled - it is not in this project however).

**Comment by cablespaghetti** [comment]
Looking at this page we may need to use the Pending status when we make the API call. Good catch!

https://developer.mailchimp.com/documentation/mailchimp/guides/manage-subscribers-with-the-mailchimp-api/

**Comment by cablespaghetti** [comment]
Yep this change definitely fixes it. :+1:

File: src/index.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ app.post('/mailsignup', function (req, res) {
         .send({
             'members': [{
                 'email_address': req.body.EMAIL,
-                'status_if_new': 'subscribed',
+                'status_if_new': 'pending',
```
Comment (jundl77) [inline src/index.js:59]: nice!

**Pull request merged**